### PR TITLE
Add automatic circle deploys to dev [SATURN-1418]

### PR DESCRIPTION
**Pull request by petesantos**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+          at: .
+      - run: cp config/<< parameters.env >>.json config.json
+      - run: echo "$<< parameters.sa_key_var >>" > /tmp/sa.json
+      - run: gcloud auth activate-service-account --key-file=/tmp/sa.json
+      - run: gcloud app deploy --project=terra-bard-<< parameters.env >> --promote --quiet
+
+jobs:
+  build:
+    docker:
+      - image: node:12
```
Comment (zarsky-broad) [inline .circleci/config.yml]: This should use the executor

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  deploy-env:
+    description: "Deploy to a specific environment"
+    parameters:
+      sa_key_var:
+        type: string
+        default: "SA_KEY_JSON"
+      env:
+        type: string
+        default: "dev"
+      pr:
```
Comment (zarsky-broad) [inline .circleci/config.yml]: Is this parameter likely to be used for this service? I suspect not.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    steps:
+      - checkout
+      - restore_cache:
+          key: v1-{{ checksum "yarn.lock" }}
+      - run: yarn install --frozen-lockfile
+      - save_cache:
+          key: v1-{{ checksum "yarn.lock" }}
+          paths:
+            - node_modules
+            - ~/.cache/yarn
```
Comment (zarsky-broad) [inline .circleci/config.yml]: Caching both has turned out to be more trouble than it's worth, IMO, in terms of how long it takes to restore all of those folders and sub-folders. I recommend just doing the `.cache`.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    docker:
+      - image: google/cloud-sdk:alpine
+
+commands:
+  deploy-env:
+    description: "Deploy to a specific environment"
+    parameters:
+      sa_key_var:
+        type: string
+        default: "SA_KEY_JSON"
```
Comment (zarsky-broad) [inline .circleci/config.yml]: Why have a default here if it's not dev?

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+      - run: yarn lint
+      - run: yarn generate-docs
+      - persist_to_workspace:
+          root: .
+          paths:
+            - .gcloudignore
+            - app.yaml
+            - config
+            - docs
+            - node_modules
```
Comment (zarsky-broad) [inline .circleci/config.yml]: I don't think you need to persist this.

File: .circleci/config.yml (hunk 2 above)
Comment (petesantos) [reply .circleci/config.yml]: If we only have dev and prod then we will not need this. 
Although, not having any service running on the other environments doesn't seem right, as we would be making requests to a non-existant service

File: .circleci/config.yml (hunk 4 above)
Comment (petesantos) [reply .circleci/config.yml]: In preparation for deployment to other environments, if we are not going to have other envs then this can be removed

**Pull request merged**